app/controllers/users_controller.rb

```diff
@@ -106,7 +106,9 @@ def index
   end
 
   def show(for_card: false)
-    return redirect_to path("/login") if SiteSetting.hide_user_profiles_from_public && !current_user
+    if SiteSetting.hide_user_profiles_from_public && !current_user
+      raise Discourse::NotFound.new(custom_message: "invalid_access", status: 403)
```

Comment:
Sorry, I didn't catch this on the first review, but could we use `Discourse::InvalidAccess` instead of `Discourse::NotFound`? Would there be a difference

Reply:
Nope, InvalidAccess adds its own redirect handling.